File: internal/api/cop/fixture.go
package cop

import (
	"context"
	"time"
)

type FixtureProvider struct {
	now func() time.Time
}

func NewFixtureProvider(now func() time.Time) *FixtureProvider {
	if now == nil {
		now = time.Now
	}
	return &FixtureProvider{now: now}
}

func (p *FixtureProvider) Snapshot(context.Context) (Snapshot, error) {
	now := p.now().UTC()
	observed := now.Add(-18 * time.Second)
	takObserved := now.Add(-46 * time.Second)
	hazardObserved := now.Add(-2 * time.Minute)
	klvObserved := now.Add(-75 * time.Second)
	sensorAltitude := 1250.5
	sensorAzimuth := 90.25
	sensorElevation := -12.5
	frameCenterElevation := 932.2

	snapshot := Snapshot{
		GeneratedAt: now,
		Scenario:    "phase-1-fixture",
		Diagnostics: SnapshotDiagnostics{
			Discovery: []DiscoveryDiagnostic{},
		},
		Feeds: []FeedHealth{
			{
				ID:          "feed.mavlink",
				Name:        "MAVLink",
				Kind:        "telemetry",
				Status:      "live",
				LastEventAt: observed,
				Message:     "Generated heartbeat and position smoke path",
			},
			{
				ID:          "feed.tak",
				Name:        "TAK/CoT",
				Kind:        "operators",
				Status:      "live",
				LastEventAt: takObserved,
				Message:     "Seed replay track, task, and GeoChat smoke path",
			},
			{
				ID:          "feed.cap",
				Name:        "CAP",
				Kind:        "advisory",
				Status:      "planned",
				LastEventAt: now.Add(-33 * time.Minute),
				Message:     "Schema/sample gate pending",
			},
			{
				ID:          "feed.klv",
				Name:        "KLV",
				Kind:        "sensor-footprint",
				Status:      "live",
				LastEventAt: klvObserved,
				Message:     "Graph-backed KLV sensor/footprint proof",
			},
			{
				ID:          "feed.weather",
				Name:        "Weather",
				Kind:        "tactical-weather",
				Status:      "planned",
				LastEventAt: now.Add(-52 * time.Minute),
				Message:     "Weather graph proof pending",
			},
		},
		Assets: []Asset{
			{
				ID:         "c360.edge.cop.mavlink.asset.system-42",
				Label:      "MAVLink system 42",
				Kind:       "mavlink-system",
				Source:     "mavlink",
				Position:   &GeoPoint{Lat: 38.9001, Lon: -77.0002},
				Confidence: 1,
				UpdatedAt:  observed,
				Provenance: Provenance{
					Owner:     "semops.feed.asset",
					SourceRef: "raw:mavlink:fixture:0001",
					Observed:  observed,
				},
			},
		},
		Tracks: []Track{
			{
				ID:         "c360.edge.cop.mavlink.track.system-42",
				Label:      "UAS 42",
				Source:     "mavlink",
				Status:     "active.armed",
				Position:   GeoPoint{Lat: 38.9001, Lon: -77.0002},
				Velocity:   "NED_CMPS(321 -12 7)",
				Confidence: 1,
				UpdatedAt:  observed,
				Provenance: Provenance{
					Owner:     "semops.feed.mavlink",
					SourceRef: "raw:mavlink:fixture:0002",
					Observed:  observed,
				},
			},
			{
				ID:         "c360.edge.cop.tak.track.android-alpha",
				Label:      "ANDROID-ALPHA",
				Source:     "tak-cot",
				Status:     "active.operator",
				Position:   GeoPoint{Lat: 38.892, Lon: -77.035},
				Velocity:   "",
				Confidence: 1,
				UpdatedAt:  takObserved,
				Provenance: Provenance{
					Owner:     "semops.feed.tak",
					SourceRef: "cot://fixture/0001",
					Observed:  takObserved,
				},
			},
		},
		Tasks: []Task{
			{
				ID:          "c360.edge.cop.tak.task.marker-north-gate",
				Label:       "North Gate",
				Kind:        "marker",
				Source:      "tak-cot",
				Status:      "active.marker",
				Position:    &GeoPoint{Lat: 38.894, Lon: -77.038},
				Description: "checkpoint",
				Confidence:  1,
				UpdatedAt:   takObserved,
				Provenance: Provenance{
					Owner:     "semops.feed.tak",
					SourceRef: "cot://fixture/0003",
					Observed:  takObserved,
				},
			},
		},
		Advisories: []Advisory{
			{
				ID:         "c360.edge.cop.tak.advisory.chat-alpha-1",
				Label:      "GeoChat ANDROID-ALPHA",
				Kind:       "geochat",
				Source:     "tak-cot",
				Status:     "active.geochat",
				Text:       "hold at checkpoint",
				Sender:     "ANDROID-ALPHA",
				Position:   &GeoPoint{Lat: 38.892, Lon: -77.035},
				Confidence: 1,
				UpdatedAt:  takObserved,
				Provenance: Provenance{
					Owner:     "semops.feed.tak",
					SourceRef: "cot://fixture/0004",
					Observed:  takObserved,
				},
			},
		},
		Hazards: []Hazard{
			{
				ID:       "c360.edge.cop.cap.hazard_area.flood-watch-1",
				Label:    "Flood watch sector",
				Kind:     "flood",
				Severity: "watch",
				Status:   "active",
				Geometry: []GeoPoint{
					{Lat: 38.895, Lon: -77.012},
					{Lat: 38.907, Lon: -77.011},
					{Lat: 38.908, Lon: -76.992},
					{Lat: 38.896, Lon: -76.991},
				},
				Source:     "cap",
				Confidence: 0.74,
				UpdatedAt:  hazardObserved,
				Provenance: Provenance{
					Owner:     "semops.feed.cap",
					SourceRef: "fixture:cap:flood-watch-1",
					Observed:  hazardObserved,
				},
			},
		},
		SensorFootprints: []SensorFootprint{
			{
				ID:             "c360.edge.cop.klv.sensor_footprint.object-semops-klv-deterministic-001-ts",
				Label:          "TEST-UAS-01 sensor footprint",
				Source:         "klv",
				Status:         "active.footprint-polygon",
				SensorPosition: GeoPoint{Lat: 38.9022, Lon: -77.0254},
				FrameCenter:    GeoPoint{Lat: 38.8956, Lon: -77.0108},
				Ray:            []GeoPoint{{Lat: 38.9022, Lon: -77.0254}, {Lat: 38.8956, Lon: -77.0108}},
				Footprint: []GeoPoint{
					{Lat: 38.8971, Lon: -77.0136},
					{Lat: 38.8968, Lon: -77.0079},
					{Lat: 38.8939, Lon: -77.0075},
					{Lat: 38.8936, Lon: -77.0132},
				},
				SensorAltitudeMeters:       &sensorAltitude,
				SensorAzimuthDegrees:       &sensorAzimuth,
				SensorElevationDegrees:     &sensorElevation,
				FrameCenterElevationMeters: &frameCenterElevation,
				MediaRef:                   "object://semops/klv/deterministic-001.ts",
				PacketRef:                  "klv://packet/deterministic/00000001",
				FrameTime:                  klvObserved,
				PlatformDesignation:        "TEST-UAS-01",
				ClaimPosture:               "sensor/frame-center and footprint polygon graph readback; tested MISB ST 0601 offset-corner subset; no STANAG conformance",
				DecodedFields: []string{
					"media_ref",
					"packet_ref",
					"observed_at",
					"platform_designation",
					"sensor_position",
					"sensor_altitude_meters",
					"sensor_azimuth_degrees",
					"sensor_elevation_degrees",
					"frame_center",
					"frame_center_elevation_meters",
					"footprint_polygon",
				},
				Warnings:   []string{},
				Confidence: 0.82,
				UpdatedAt:  klvObserved,
				Provenance: Provenance{
					Owner:     "semops.feed.klv",
					SourceRef: "klv://packet/deterministic/00000001",
					Observed:  klvObserved,
				},
			},
		},
		Weather: []WeatherObservation{},
		Alerts: []Alert{
			{
				ID:        "alert.mavlink.track-freshness",
				Label:     "Track freshness nominal",
				Severity:  "info",
				Status:    "active",
				EntityID:  "c360.edge.cop.mavlink.track.system-42",
				Reason:    "MAVLink position observed within freshness window",
				UpdatedAt: observed,
			},
		},
	}
	snapshot.Summary = Summary{
		ActiveTracks:           len(snapshot.Tracks),
		ActiveTasks:            len(snapshot.Tasks),
		ActiveAdvisories:       len(snapshot.Advisories),
		ActiveSensorFootprints: len(snapshot.SensorFootprints),
		ActiveWeather:          len(snapshot.Weather),
		ActiveAlerts:           len(snapshot.Alerts),
		StaleFeeds:             countFeeds(snapshot.Feeds, "stale"),
	}
	return snapshot, nil
}

func countFeeds(feeds []FeedHealth, status string) int {
	var count int
	for _, feed := range feeds {
		if feed.Status == status {
			count++
		}
	}
	return count
}

File: ui/src/lib/cop/fixture.ts
import type { Snapshot } from './types';

const now = '2026-06-19T12:00:00Z';
const observed = '2026-06-19T11:59:42Z';
const takObserved = '2026-06-19T11:59:14Z';
const advisoryObserved = '2026-06-19T11:58:00Z';
const klvObserved = '2026-06-19T11:58:45Z';
const weatherObserved = '2026-06-19T11:57:30Z';
const commandObserved = '2026-06-19T11:59:30Z';
const adsbObserved = '2026-06-19T11:59:35Z';
const fusionObserved = '2026-06-19T11:59:36Z';

export const fixtureSnapshot: Snapshot = {
  generated_at: now,
  scenario: 'phase-1-fixture',
  summary: {
    active_tracks: 3,
    active_tasks: 2,
    active_advisories: 1,
    active_sensor_footprints: 1,
    active_weather_observations: 1,
    active_associations: 1,
    active_alerts: 1,
    stale_feeds: 0
  },
  diagnostics: {
    discovery: []
  },
  feeds: [
    {
      id: 'feed.mavlink',
      name: 'MAVLink',
      kind: 'telemetry',
      status: 'live',
      last_event_at: observed,
      message: 'Generated heartbeat and position smoke path'
    },
    {
      id: 'feed.tak',
      name: 'TAK/CoT',
      kind: 'operators',
      status: 'live',
      last_event_at: takObserved,
      message: 'Seed replay track, task, and GeoChat smoke path'
    },
    {
      id: 'feed.command',
      name: 'Command',
      kind: 'control',
      status: 'live',
      last_event_at: commandObserved,
      message: 'Synthetic command lifecycle replay state'
    },
    {
      id: 'feed.cap',
      name: 'CAP',
      kind: 'advisory',
      status: 'planned',
      last_event_at: '2026-06-19T11:27:00Z',
      message: 'Schema/sample gate pending'
    },
    {
      id: 'feed.adsb',
      name: 'ADS-B',
      kind: 'air-picture',
      status: 'live',
      last_event_at: adsbObserved,
      message: 'Fixture-backed ADS-B aircraft track'
    },
    {
      id: 'feed.klv',
      name: 'KLV',
      kind: 'sensor-footprint',
      status: 'live',
      last_event_at: klvObserved,
      message: 'Graph-backed KLV sensor/frame-center proof'
    },
    {
      id: 'feed.weather',
      name: 'Weather',
      kind: 'tactical-weather',
      status: 'live',
      last_event_at: weatherObserved,
      message: 'Fixture-backed point weather observation readback'
    },
    {
      id: 'feed.fusion',
      name: 'Fusion',
      kind: 'inference',
      status: 'live',
      last_event_at: fusionObserved,
      message: 'Fixture-backed fusion association evidence'
    }
  ],
  assets: [
    {
      id: 'c360.edge.cop.mavlink.asset.system-42',
      label: 'MAVLink system 42',
      kind: 'mavlink-system',
      source: 'mavlink',
      position: { lat: 38.9001, lon: -77.0002 },
      confidence: 1,
      updated_at: observed,
      provenance: {
        owner: 'semops.feed.asset',
        source_ref: 'raw:mavlink:fixture:0001',
        observed_at: observed
      }
    }
  ],
  tracks: [
    {
      id: 'c360.edge.cop.mavlink.track.system-42',
      label: 'UAS 42',
      source: 'mavlink',
      status: 'active.armed',
      position: { lat: 38.9001, lon: -77.0002 },
      velocity: 'NED_CMPS(321 -12 7)',
      confidence: 1,
      updated_at: observed,
      provenance: {
        owner: 'semops.feed.mavlink',
        source_ref: 'raw:mavlink:fixture:0002',
        observed_at: observed
      }
    },
    {
      id: 'c360.edge.cop.tak.track.android-alpha',
      label: 'ANDROID-ALPHA',
      source: 'tak-cot',
      status: 'active.operator',
      position: { lat: 38.892, lon: -77.035 },
      velocity: '',
      confidence: 1,
      updated_at: takObserved,
      provenance: {
        owner: 'semops.feed.tak',
        source_ref: 'cot://fixture/0001',
        observed_at: takObserved
      }
    },
    {
      id: 'c360.edge.cop.adsb.track.a1b2c3',
      label: 'N42CX',
      source: 'adsb',
      status: 'active.aircraft',
      position: { lat: 38.90028, lon: -77.00005 },
      velocity: 'GROUND_KTS(73)',
      confidence: 0.88,
      updated_at: adsbObserved,
      provenance: {
        owner: 'semops.feed.adsb',
        source_ref: 'adsb://opensky/fixture/a1b2c3',
        observed_at: adsbObserved
      }
    }
  ],
  tasks: [
    {
      id: 'c360.edge.cop.tak.task.marker-north-gate',
      label: 'North Gate',
      kind: 'marker',
      source: 'tak-cot',
      status: 'active.marker',
      position: { lat: 38.894, lon: -77.038 },
      description: 'checkpoint',
      confidence: 1,
      updated_at: takObserved,
      provenance: {
        owner: 'semops.feed.tak',
        source_ref: 'cot://fixture/0003',
        observed_at: takObserved
      }
    },
    {
      id: 'c360.edge.cop.command.task.csapi-command-route-42',
      label: 'Route MAVLink system 42 to North Gate',
      kind: 'mavlink.goto',
      source: 'command',
      status: 'cancel_requested',
      description: 'cancel requested: airspace conflict',
      target_id: 'c360.edge.cop.mavlink.asset.system-42',
      authority: 'local.operator',
      priority: 95,
      expires_at: '2026-06-19T12:03:00Z',
      requested_by: 'operator:lead',
      correlation_id: 'ui:cancel-route-42',
      desired_state: '{"command":"cancel","target_native_id":"csapi-command-route-42","reason":"airspace conflict"}',
      confidence: 1,
      updated_at: commandObserved,
      provenance: {
        owner: 'semops.command.intent',
        source_ref: 'command://fixture/hadr-command/0004-route-cancel-requested',
        observed_at: commandObserved
      }
    }
  ],
  advisories: [
    {
      id: 'c360.edge.cop.tak.advisory.chat-alpha-1',
      label: 'GeoChat ANDROID-ALPHA',
      kind: 'geochat',
      source: 'tak-cot',
      status: 'active.geochat',
      text: 'hold at checkpoint',
      sender: 'ANDROID-ALPHA',
      position: { lat: 38.892, lon: -77.035 },
      confidence: 1,
      updated_at: takObserved,
      provenance: {
        owner: 'semops.feed.tak',
        source_ref: 'cot://fixture/0004',
        observed_at: takObserved
      }
    }
  ],
  hazards: [
    {
      id: 'c360.edge.cop.cap.hazard_area.flood-watch-1',
      label: 'Flood watch sector',
      kind: 'flood',
      severity: 'watch',
      status: 'active',
      geometry: [
        { lat: 38.895, lon: -77.012 },
        { lat: 38.907, lon: -77.011 },
        { lat: 38.908, lon: -76.992 },
        { lat: 38.896, lon: -76.991 }
      ],
      source: 'cap',
      confidence: 0.74,
      updated_at: advisoryObserved,
      provenance: {
        owner: 'semops.feed.cap',
        source_ref: 'fixture:cap:flood-watch-1',
        observed_at: advisoryObserved
      }
    }
  ],
  sensor_footprints: [
    {
      id: 'c360.edge.cop.klv.sensor_footprint.object-semops-klv-deterministic-001-ts',
      label: 'TEST-UAS-01 sensor footprint',
      source: 'klv',
      status: 'active.sensor-frame-center',
      sensor_position: { lat: 38.9022, lon: -77.0254 },
      frame_center: { lat: 38.8956, lon: -77.0108 },
      ray: [
        { lat: 38.9022, lon: -77.0254 },
        { lat: 38.8956, lon: -77.0108 }
      ],
      sensor_altitude_meters: 1250.5,
      sensor_azimuth_degrees: 90.25,
      sensor_elevation_degrees: -12.5,
      frame_center_elevation_meters: 932.2,
      media_ref: 'object://semops/klv/deterministic-001.ts',
      packet_ref: 'klv://packet/deterministic/00000001',
      frame_time: klvObserved,
      platform_designation: 'TEST-UAS-01',
      claim_posture: 'sensor-frame-center graph readback; no footprint polygon; no STANAG conformance',
      decoded_fields: [
        'media_ref',
        'packet_ref',
        'observed_at',
        'platform_designation',
        'sensor_position',
        'sensor_altitude_meters',
        'sensor_azimuth_degrees',
        'sensor_elevation_degrees',
        'frame_center',
        'frame_center_elevation_meters'
      ],
      warnings: ['footprint polygon not computed'],
      confidence: 0.82,
      updated_at: klvObserved,
      provenance: {
        owner: 'semops.feed.klv',
        source_ref: 'klv://packet/deterministic/00000001',
        observed_at: klvObserved
      }
    }
  ],
  weather_observations: [
    {
      id: 'c360.edge.cop.weather.weather_observation.open-meteo-position-fixture-temperature-2m',
      label: '29.4 degC temperature_2m',
      source: 'weather',
      status: 'fresh',
      provider: 'open-meteo',
      query_shape: 'position',
      query_geometry_wkt: 'POINT(-77.0400000 38.9000000)',
      position: { lat: 38.9, lon: -77.04 },
      valid_time: weatherObserved,
      model_time: '2026-06-19T11:00:00Z',
      fresh_until: '2026-06-19T12:57:30Z',
      variable: 'temperature_2m',
      value: 29.4,
      unit: 'degC',
      claim_posture: 'fixture-backed point observation; no live provider, weather tile, route-safety, or OGC conformance claim',
      confidence: 0.78,
      updated_at: weatherObserved,
      provenance: {
        owner: 'semops.feed.weather',
        source_ref: 'weather://open-meteo/fixture/position/temperature_2m/2026-06-19T11:57:30Z',
        observed_at: weatherObserved
      }
    }
  ],
  associations: [
    {
      id: 'c360.edge.cop.fusion.association.mavlink-to-adsb',
      label: 'Track association UAS 42 -> N42CX ambiguous',
      kind: 'track',
      source: 'fusion',
      status: 'ambiguous',
      primary_track_id: 'c360.edge.cop.mavlink.track.system-42',
      candidate_track_id: 'c360.edge.cop.adsb.track.a1b2c3',
      algorithm: 'semops.association.geotemporal.v1',
      reason: 'sources=mavlink,adsb; distance_meters=24; time_delta_seconds=7.000; ambiguous_with=adsb-altitude-shadow',
      distance_meters: 24,
      time_delta_seconds: 7,
      claim_posture: 'fusion association evidence; no source-track merge; no identity authority',
      confidence: 0.87,
      updated_at: fusionObserved,
      provenance: {
        owner: 'semops.fusion.structural',
        source_ref: 'primary=mavlink://raw/udp/0002 candidate=adsb://opensky/fixture/a1b2c3',
        observed_at: fusionObserved
      }
    }
  ],
  alerts: [
    {
      id: 'alert.mavlink.track-freshness',
      label: 'Track freshness nominal',
      severity: 'info',
      status: 'active',
      entity_id: 'c360.edge.cop.mavlink.track.system-42',
      reason: 'MAVLink position observed within freshness window',
      updated_at: observed
    }
  ]
};
